### PR TITLE
Replace fetch with axios (Resolves #9)

### LINKED ISSUE #9: Axios
I'd like to suggest the use of [axios](https://github.com/mzabriskie/axios) over fetch. It has quite a few benefits over fetch.

- Global Auth (Meaning auth would not be a concern of this package, you'd pass through an axios client preconfigured for auth)
- Interceptors (For an example of why this is useful, I use OAuth for API authentication, so the client deals with hourly access_token refreshes, and needs to handle this transparently, an interceptor allows me to automatically catch 403's and refresh the access_token and re-fire the request with these. I don't have an example Axios implementation, but I do have one for Angular's $http [here](https://github.com/Poniverse/Poniverse.net/blob/master/src/client/app/blocks/recover-session/recover-session.js) which is very similar)
- Generally allows for much more control by the implementation of this package.
- `.then` and `.catch` are properly fired based on the response code, so no more manual status code checking.

If this would be something you're interested in, I'd be happy to fork and add it myself.

### PR DESCRIPTION
**Pull request by Lavoaster**
Sorry it's taken so long, but here it is! 🎉 

Added thanks to @BenBarber for doing some additional work around this.

### CODE REVIEW

**Comment by egeriis** [comment]
Thanks @Lavoaster! We'll take a look and review your PR in the coming days.

File: src/jsonapi.js [hunk 1]
```diff
@@ -49,34 +48,43 @@ export const uploadFile = (file, {
...
+    //     const e = new Error(res.statusText);
+    //     e.response = res;
+    //     throw e;
+    //   })
+    //   .then(json => {
+    //     onSuccess(json);
+    //   })
+    //   .catch(error => {
+    //     onError(error);
+    //   });
```
Comment (egeriis) [inline src/jsonapi.js]: There's a bunch of out-commented code here

File: src/jsonapi.js [hunk 2]
```diff
@@ -49,34 +48,43 @@ export const uploadFile = (file, {
...
+
+    return apiRequest(endpoint, options)
       .then(json => {
-        onSuccess(json);
+        onSuccess(res.data);
```
Comment (egeriis) [inline src/jsonapi.js]: `res` is undefined?

File: src/utils.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import axios from 'axios';
+
 export const jsonContentTypes = [
```
Comment (egeriis) [inline src/utils.js]: This variable does not seem to be used anymore

File: src/jsonapi.js [hunk 4]
```diff
@@ -49,34 +48,43 @@ export const uploadFile = (file, {
...
+    //     throw e;
+    //   })
+    //   .then(json => {
+    //     onSuccess(json);
+    //   })
+    //   .catch(error => {
+    //     onError(error);
+    //   });
+
+    return apiRequest(endpoint, options)
```
Comment (egeriis) [inline src/jsonapi.js]: `options` is provided as second argument, which is supposed to be `accessToken`. It does appear that `accessToken` is no longer used, though.

**Comment by egeriis** [comment]
Hey @Lavoaster. Finally got the time to review your PR. There's only a few questions and notes:

1. I was considering that one of the benefits of using Axios, would be that endpoint host and path does not need to be configured through the Redux store and this library's reducers. Did you consider this and is it feasible?
2. Will `res.data` automatically become a JSON object when a JSON Content-Type header is provided in the response?
3. We should make an update to the README, with an updated implementation guide.

**Comment by egeriis** [comment]
We need to have this PR resolved, or we'll move along with another solution. @Lavoaster Can you let us know if you plan to review our feedback in the near future? Thanks.

**Comment by asiniy** [comment]
@egeriis I can start to work on this PR a couple of days later if @Lavoaster is quiet

**Comment by egeriis** [comment]
@asiniy Sounds good. Any contribution is very welcome :)

**Comment by Lavoaster** [comment]
@asiniy Feel free to pick up on this.

@egeriis I heavily apologise for the lack of a reply  / update here, this package at was dropped at my workplace in favour of rolling our own. (Not to say that this package doesn't do a good job) I won't be able to continue work on this.

**Comment by egeriis** [comment]
@Lavoaster Thank you for following up. Can I ask you if you identified some lack of features or what was the rationale behind writing your own state management?

**Pull request closed without merging**